### PR TITLE
nixos: Fix certificateKey and certificateFile handling

### LINKED ISSUE #112: error: A definition for option `services.tsnsrv.defaults.certificateFile' is not of type `path'
I suspect #106 broke non-custom-TLS configuration:

```
warning: Git tree '/Users/asf/home' is dirty
copying path '/nix/store/kii30vrsz832yg94v9w2imlwn8pp69xq-source' from 'https://cache.nixos.org'...
error:
       … while calling the 'head' builtin

         at /nix/store/4y2xyaq03yjkzh2rp7yqcwikznns72pl-source/lib/attrsets.nix:922:11:

          921|         || pred here (elemAt values 1) (head values) then
          922|           head values
             |           ^
          923|         else

       … while evaluating the attribute 'value'

         at /nix/store/4y2xyaq03yjkzh2rp7yqcwikznns72pl-source/lib/modules.nix:807:9:

          806|     in warnDeprecation opt //
          807|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          808|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: A definition for option `services.tsnsrv.defaults.certificateFile' is not of type `path'. Definition values:
       - In `/nix/store/42f0fibk91ld4pfg1irad7x8rpvwj0ja-source/flake.nix#nixosModules.default': ""
```

I think we should have tests for those nixos modules... but let's fix this first.

### PR DESCRIPTION
**Pull request by antifuchs**
Fixes #112 

* service.certificateKey was referred to as service.keyFile, which is an option that doesn't exist (but does exist when acmeHost is set). Changed to service.certificateKey

* The certificateFile and certificateKey options were not nullable, meaning they were required to be set (and would cause tsnsrv to use a manual cert/key if they weren't and things worked). Now they're nullable, defaulting to null, and only get passed to tsnsrv if both are set.

### CODE REVIEW

**Comment by antifuchs** [comment]
@SuperSandro2000 - I think this change should still correctly handle non-default certificate handling. Can you check to make sure I don't break your stuff? (:

File: nixos/default.nix [hunk 1]
```diff
@@ -169,6 +167,13 @@
     ]
     ++ lib.optionals (service.whoisTimeout != null) ["-whoisTimeout" service.whoisTimeout]
     ++ lib.optionals (service.upstreamUnixAddr != null) ["-upstreamUnixAddr" service.upstreamUnixAddr]
+    ++ lib.optionals (
+      (lib.assertMsg (((service.certificateFile != null) && (service.certificateKey != null)) || ((service.certificateFile == null) && (service.certificateKey == null))) "Both certificateFile and certificateKey must be set or null on services.tsnsrv.${name}")
```
Comment (SuperSandro2000) [inline nixos/default.nix:171]: This should be in assertions to give proper module errors

Comment (antifuchs) [reply nixos/default.nix]: That's fantastic, I think I knew about these once but forgot they exist. Much nicer this way.

**Comment by SuperSandro2000** [comment]
I've rebased #111 on top of this and it didn't even trigger a service restart. Good job!

**Pull request merged**